pkg/runtime/runtime_manifest.go

```diff
@@ -271,9 +297,12 @@ func (m *RuntimeManifest) buildSourceData() (map[string]interface{}, error) {
 		}
 
 		pe := param.Name
-		val := m.resolveParameter(param)
+		val, err := m.resolveParameter(param)
+		if err != nil {
+			return nil, err
+		}
 		if param.Sensitive {
-			m.setSensitiveValue(val)
+			m.setSensitiveValue(val.(string))
```

Comment:
One possible solution to only handling `string`s would be to do a `fmt.Sprint(val)` here, since that will invoke the `Format` function of object-type parameters and make them "strings".

Reply:
Will this mean that object properties cannot be marked as sensitive, and still work? Could you add a test that it works for sensitive values?

Reply:
Ok, that makes sense. Could you create an issue for this so we don't forget?
And thank you very much for the work on this

Reply:
Created https://github.com/getporter/porter/issues/3227.

Reply:
Thank you very much